pkg/tbtc/tbtc.go

```diff
@@ -125,9 +136,39 @@ func Initialize(
 		}()
 	})
 
-	// TODO: This is a temporary signing loop trigger that should be removed
-	//       once the client is integrated with real on-chain contracts.
-	_ = chain.OnSignatureRequested(func(event *SignatureRequestedEvent) {
+	_ = chain.OnDKGResultSubmitted(func(event *DKGResultSubmittedEvent) {
+		go func() {
+			if ok := deduplicator.notifyDKGResultSubmitted(
+				event.ResultHash,
+			); !ok {
+				logger.Warnf(
+					"DKG result with hash [0x%x] and starting "+
+						"block [%v] has been already processed",
+					event.ResultHash,
+					event.BlockNumber,
+				)
+				return
+			}
+
+			logger.Infof(
+				"DKG result with hash [0x%x] submitted at block [%v]",
+				event.ResultHash,
+				event.BlockNumber,
+			)
+
+			node.validateDKG(
+				event.Seed,
+				event.BlockNumber,
+				event.Result,
+				event.ResultHash,
+			)
+		}()
+	})
+
+	// TODO: Repeat DKG in case of the result challenge and archive the
+	//       signers stored for the invalid result.
```

Comment:
See https://github.com/keep-network/keep-core/pull/3427#discussion_r1053377999. Regarding:

> The risk with the key generation happening more than once is that someone can try to submit the result for another DKG that happens in the future.

I don't quite get it. Can you write a problematic scenario?

Reply:
I was not super precise here: if we allow for another DKG to happen, then we end up with two valid DKG results and both of them can be submitted and pass the on-chain validation. We can continue this discussion in [this](https://github.com/keep-network/keep-core/pull/3427#discussion_r1055513333) thread as it's basically about the same concerns. I'll leave this comment open so that we can remember about potentially updating/removing the TODO.

Reply:
Yeah, that was cleared out in the referenced discussion. I removed that TODO (9ea6adf4) since we have this work captured by https://github.com/keep-network/keep-core/issues/3450